modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -227,6 +227,8 @@ object CrossSources {
       lazy val dir     = sourcePath / os.up
       lazy val subPath = sourcePath.subRelativeTo(dir)
       if (os.isDir(sourcePath)) Right(Inputs.singleFilesFromDirectory(Inputs.Directory(sourcePath)))
+      else if (sourcePath.endsWith(os.SubPath("project.settings.scala")))
```

Comment:
Do we want accept also files such as `Hello.project.settings.scala`?
```suggestion
      else if (sourcePath == os.SubPath("project.settings.scala"))
```